js/client-topbar.js

```diff
@@ -1010,10 +1010,21 @@
 				buf += '<div id="gapi-custom-signin" style="width:240px;margin:0 auto">[loading Google log-in button]</div>';
 				buf += '<p class="buttonbar"><button name="close">Cancel</button></p>';
 			} else {
-				buf += '<p><label class="label">Password: <input class="textbox autofocus" type="password" name="password" autocomplete="current-password"></label></p>';
+				buf += '<p><label class="label">Password: <input class="textbox autofocus" type="password" name="password" id="pwd" autocomplete="current-password"></label><i class="fa fa-eye-slash showpassword" onclick="showPassword(this)"></i></p>';
 				buf += '<p class="buttonbar"><button type="submit"><strong>Log in</strong></button> <button name="close">Cancel</button></p>';
 			}
 
+			function showPassword(fa) {
```

Comment:
This is not a global; if it works, I'd be very wary of how magical it is.

We have a pseudo-framework for this; you should be using `<button name="showPassword">` and then define a method `showPassword:` like how `initialize:` is defined.